shotover-proxy/src/message/mod.rs

```diff
@@ -195,6 +202,7 @@ impl Message {
             MessageInner::Modified { frame } | MessageInner::Parsed { frame, .. } => match frame {
                 Frame::Cassandra(frame) => frame.cell_count()?,
                 Frame::Redis(_) => nonzero!(1u32),
+                Frame::Kafka(_) => todo!(),
```

Comment:
Couldn't this just be `nonzero(1u32)` same as Redis?

https://dattell.com/data-architecture-blog/load-balancing-with-kafka/#:~:text=What%20is%20Kafka%20loading%20balancing,partitions%20while%20preserving%20message%20ordering.

From a quick read I think it's safe to just assume 1  kafka message = 1 throttling cell

Reply:
in cassandra we count each message within a batch as a separate cell, kafka produce messages allow for [batching many messages together](https://kafka.apache.org/protocol.html#The_Messages_Produce) so it feels equivalent to a cassandra batch.
Maybe its closer to just having multiple rows on an insert though?
Its a complicated question which I considered it out of scope and left as a todo!()

Reply:
Ah good point, didn't know that. Thanks